### DISCUSSION TITLE
Enhancement: Link to specific modelSpec

### DISCUSSION THREAD

**Discussion by cmcfadden** [danny-avila/LibreChat#2726] [Feature Requests & Suggestions]
### What features would you like to see added?

I'd like to be able to build a landing page with descriptions of a variety of modelSpecs (outside of LibreChat), then link to individual options. That will make it easier for folks to discover the different modelSpecs, and allow me to guide users to the right destination. 

### More details

I have a branch where I made this settable with a hash parameter in the URL (`#modelSpec=<spec.name>`). I'm happy to do this as a PR, just wasn't sure if this is an option the authors would be open to. 

```
export function getDefaultModelSpec(modelSpecs?: TModelSpec[]) {
  const hashSpecName = new URLSearchParams(window.location.hash).get('#modelSpec');
  const hashSpec = modelSpecs?.find((spec) => spec.name === hashSpecName);
  if(hashSpec) {
    window.location.hash = '';
  }
  const defaultSpec = modelSpecs?.find((spec) => spec.default);
  const lastSelectedSpecName = localStorage.getItem(LocalStorageKeys.LAST_SPEC);
  const lastSelectedSpec = modelSpecs?.find((spec) => spec.name === lastSelectedSpecName);
  return hashSpec || defaultSpec || lastSelectedSpec || modelSpecs?.[0];
}
```

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
so you can do this with URL params? For the landing page variation, as long as this is configurable and not the default option, I'm open to this

**Comment by cmcfadden** [comment]
I've been playing with this a bit more, and I'm considering alternative approaches - the downside of this is approach is that it'll get lost during an auth flow. Could potentially have a url that would update the local storage so that it was persistent prior to the auth flow? Hmm.

**Reply by danny-avila** [reply]
is this for external links from another app/site, or within the app?

**Reply by cmcfadden** [reply]
What I'm interested in is having a landing page which highlights different model/preset combos, for specific purposes. Like a grid of icons for example, allowing you to link in to a specific model/preset.

**Reply by danny-avila** [reply]
I see, in this case, it's really best to use the existing logic for selecting presets/modelSpecs. It is a bit complex due to the flexibility/variation between endpoints but it really could be the same as examples from the header options to make this easy.
